libsignal-protocol/src/errors.rs

```diff
@@ -112,6 +114,22 @@ impl InternalErrorCode for i32 {
             Some(e) => Err(e),
         }
     }
+
+    fn into_code(self) -> i32 { self }
+}
+
+impl<T> InternalErrorCode for Result<T, InternalError>
+where
+    T: Any,
+{
+    fn into_result(self) -> Result<(), InternalError> { self.map(|_| ()) }
+
+    fn into_code(self) -> i32 {
```

Comment:
I think this should actually be pulled out into its own trait (maybe `IntoInternalErrorCode` and rename `InternalErrorCode` to `FromInternalErrorCode`). 

The `into_result()` function is `i32 -> Result<(), InternalError>`, whereas `into_code()` is `Result<T, InternalError> -> i32`. It's a bit like the result between `std::iter::IntoIterator` and `std::iter::FromIterator`... if that makes sense?

Reply:
Yup, it makes sense, also maybe make a blanket implementation for every type that implements one of them to the other, as i know that was the case for the `From` and `Into` traits.